src/utils/localization.ts

```diff
@@ -0,0 +1,21 @@
+import { Locale, Localization } from "../types";
+
+interface LocalizationObject {
+  localization: Localization;
+}
+type LocalizedObject<T extends string> = LocalizationObject & {
+  [key in T]: string;
+};
+// interface LocalizedObject<T> extends { localization: Localization; };
+
+export const getLocalizedString = <T extends string>({
+  obj,
+  property,
+  locale,
+  fallback,
+}: {
+  obj?: LocalizedObject<T>;
+  property: T;
+  locale: Locale;
+  fallback: string;
+}): string => obj?.localization[locale]?.description ?? (obj && obj[property]) ?? fallback;
```

Comment:
I mean now it's part of whoever wants to localize a string to set a fallback for it, which in these cases would be the original english string, and fallback is always required so it is needed to provide that string

Reply:
We could add it here, but we'd need a fallback object with all the fallbacks, and then if a fallback is not provided we default to English. I think that makes sense in order to have it centralised.

Reply:
Isnt the english text defaulted in `obj && property && obj[property]` ? From what I see in yearn-meta, that is how it has been structured, and should always exist.

Reply:
Or you mean something like:
```
obj?.localization[locale]?.description ?? obj?.localization['en']?.description ?? (obj && obj[property]) ?? fallback
```
?

Reply:
I think what @jstashh was referring to is that we shouldn't need a required `fallback` prop and it would just fallback to English in the SDK. If the `fallback` prop is given then it would override the default fallback.

Reply:
not a blocker though - let's merge this to move forwards

Reply:
ok, will merge then, and if it becomes an issue we will revisit it